tests/foreman/cli/test_variables.py

```diff
@@ -16,122 +16,185 @@
 @Upstream: No
 """
 
+from robottelo import ssh
+from robottelo.cli.base import CLIReturnCodeError
+from robottelo.cli.environment import Environment
+from robottelo.cli.factory import make_hostgroup, make_variable
+from robottelo.cli.host import Host
+from robottelo.cli.hostgroup import HostGroup
+from robottelo.cli.proxy import Proxy
+from robottelo.cli.puppet import Puppet
+from robottelo.cli.variable import SmartVariable
+from robottelo.config import settings
+from robottelo.datafactory import (
+    gen_string,
+    invalid_values_list,
+    valid_data_list,
+)
 from robottelo.decorators import (
     run_only_on,
+    skip_if_bug_open,
     stubbed,
     tier1,
     tier2,
-    tier3
 )
 from robottelo.test import CLITestCase
 
 
 class SmartVariablesTestCase(CLITestCase):
     """Implements Smart Variables tests in CLI"""
 
+    @classmethod
+    def setUpClass(cls):
+        """Import some parametrized puppet classes. This is required to make
+        sure that we have data to be able to perform interactions with smart
+        class variables.
+        """
+        super(SmartVariablesTestCase, cls).setUpClass()
+        cls.host_name = settings.server.hostname
+        ssh.command('puppet module install --force puppetlabs/ntp')
+        cls.env = Environment.info({u'name': 'production'})
+        Proxy.importclasses({
+            u'environment': cls.env['name'],
+            u'name': cls.host_name,
+        })
+        cls.puppet = Puppet.info({u'name': 'ntp'})
+
     @run_only_on('sat')
-    @stubbed()
     @tier2
     def test_positive_list_variables_by_host_name(self):
         """List all smart variables associated to host by hostname.
 
         @id: ee0da54c-ab60-4dde-8e1f-d548b52bac73
 
-        @steps: List all the smart variables associated to specific host by
-        hostname.
-
         @assert: Smart Variables listed for specific Host by hostname.
 
         @CaseLevel: Integration
         """
+        make_variable({
+            'variable': gen_string('alpha'),
```

Comment:
nitpick: factory will populate random name (as you defined it in make_variable), so you may drop this and all the similar lines in other tests.